### PR TITLE
PHPDoc cleanup

### PR DESCRIPTION
**Pull request by demis-palma**
Some PHPDoc comments lack parameters, or indicate wrong type or number of parameters.
Others doesn't contain all necessary @throws or @return tags

To be merged on review.

### CODE REVIEW

File: libraries/joomla/filesystem/file.php [hunk 1]
```diff
@@ -512,7 +512,7 @@ public static function append($file, $buffer, $use_streams = false)
 	 * @param   string   $dest             The path (including filename) to move the uploaded file to
 	 * @param   boolean  $use_streams      True to use streams
 	 * @param   boolean  $allow_unsafe     Allow the upload of unsafe files
-	 * @param   boolean  $safeFileOptions  Options to JFilterInput::isSafeFile
+	 * @param   array  $safeFileOptions  Options to JFilterInput::isSafeFile
```
Comment (brianteeman) [inline libraries/joomla/filesystem/file.php]: could you fix the alignment here please

File: libraries/joomla/form/form.php [hunk 2]
```diff
@@ -704,9 +704,9 @@ public function renderFieldset($name, $options = array())
 	 * method will move on to the next field to load.
 	 *
 	 * @param   string  $data     The name of an XML string or object.
-	 * @param   string  $replace  Flag to toggle whether form fields should be replaced if a field
+	 * @param   bool  $replace  Flag to toggle whether form fields should be replaced if a field
```
Comment (brianteeman) [inline libraries/joomla/form/form.php]: could you fix the alignment here please

File: libraries/joomla/form/form.php [hunk 3]
```diff
@@ -828,9 +828,9 @@ public function load($data, $replace = true, $xpath = false)
 	 * to false.
 	 *
 	 * @param   string  $file   The filesystem path of an XML file.
-	 * @param   string  $reset  Flag to toggle whether form fields should be replaced if a field
+	 * @param   bool  $reset  Flag to toggle whether form fields should be replaced if a field
```
Comment (brianteeman) [inline libraries/joomla/form/form.php]: could you fix the alignment here please

**Comment by brianteeman** [comment]
Can you remove the non php document changes from this PR and put them in their own PR please

**Comment by demis-palma** [comment]
@brianteeman alignment done.
About non PHP document changes, If you refer to the parameter $startTime = 0.0 I did it to fix another inconsistency between the doc and the code.

The PHPdoc declares the variable as "double"
`@param   double  $startTime`
but the method initialises it with an "integer" value
`$startTime = 0`

gettype(0) returns "integer"
gettype(0.0) returns "double"

As far as I know, the correct way to initialise a "double/float" without using a cast, is using the decimal point.

But if you have other reasons, I can separate this last.

**Comment by RonakParmar** [comment]
I have tested this item :white_check_mark: successfully on c0a4b79d4148528b01e7d4cfc0c94288914a9d65<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/15702">issues.joomla.org/tracker/joomla-cms/15702</a>.</sub>

**Comment by brianteeman** [comment]
A pull request should have a single responsibility.

**Comment by demis-palma** [comment]
In this case, profiler restored.

File: libraries/joomla/google/data/picasa.php [hunk 4]
```diff
@@ -92,7 +92,7 @@ public function listAlbums($userID = 'default')
 	 *
 	 * @since   12.3
 	 */
-	public function createAlbum($userID = 'default', $title = '', $access = 'private', $summary = '', $location = '', $time = false, $keywords = array())
+	public function createAlbum($userID = 'default', $title = '', $access = 'private', $summary = '', $location = '', $time = null, $keywords = array())
```
Comment (Quy) [inline libraries/joomla/google/data/picasa.php]: Revert this?

Comment (demis-palma) [inline libraries/joomla/google/data/picasa.php]: Done.

**Comment by jonasgonka** [comment]
Because some of the files were moved I updated this and created a new PR #21234.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/15702">issues.joomla.org/tracker/joomla-cms/15702</a>.</sub>

**Comment by ghost** [comment]
can this PR be closed?

**Comment by jonasgonka** [comment]
this PR can be closed.

**Pull request closed without merging**